### DISCUSSION TITLE
Can LTDC directly operate with display without driver chip?

### DISCUSSION THREAD

**Discussion by rtek1000** [stm32duino/Arduino_Core_STM32#1624] [Libraries] [answered]
Hello,

I know that LTDC can operate with display with driver chip, like ILI9341. But can it operate with display without the driver chip?

There are many display models without a driver chip, and many have datasheets. They are cheaper than driver chip displays.

The AT070TN94 display needs constant updating, I have boards with the S3C2440 microcontroller that work directly with this type of display.

I found some setup information on the S3C2440 from a similar display:

https://www.programmerall.com/article/5735137131/

But I didn't find much about this type of display and STM32.

I found this post below, but there was no response:

https://community.st.com/s/question/0D50X0000BqC2xe/7inch-at070tn94-lcd-and-stm32f429-on-stm32cube-setup-on-custom-board

Thank you.

**Comment by ABOSTM** [comment]
Hi @rtek1000 ,
I am not an LCD expert, but this Application Note should help you: AN4861
https://www.st.com/resource/en/application_note/dm00287603-lcdtft-display-controller-ltdc-on-stm32-mcus-stmicroelectronics.pdf

Especially  `1.1 Basic graphics concepts `, `§1.2 Display interface standards`  and `§1.3 Display interfaces supported by STM32 MCU_s`
It is said that:
> Here below a summary on the MIPI Alliance display interfaces supported by STM32 MCUs:
> • All STM32 MCUs support the MIPI-DBI type C (SPI) interface.
> • All STM32 MCUs with F(S)MC support the MIPI-DBI type A and B interfaces.
> • STM32 MCUs with **LTDC** support the MIPI-DPI interface.
> • STM32 MCUs embedding a DSI Host support the MIPI-DSI interface.

As far as I understand, LTDC allows to control LCD without controller, thanks to RGB parallel interface
![image](https://user-images.githubusercontent.com/50730894/149967482-55f79cf0-bbb0-40b1-815b-f23b70d97946.png)

It seems to be the case of AT070TN94

**Comment by rtek1000** [comment]
Hi,

Yes, I saw this document yesterday, and I understood that it can work with LCD without a driver chip, but...

I think a lot of people don't know how to use it properly, and they are paying too much for displays with a driver chip and even reinforcing the use of modular Nextion-type displays.

From what I understand from this AN, care must be taken to choose the correct clock to match the display clock, and it must also be necessary to use an external RAM memory as a buffer for the display frame.

Therefore, it may be more advantageous to use a driver chip, to avoid a lot of setup with external memory. Since it's not just a software issue, but also the hardware doesn't have all the support to perform the direct operation with the TFT display (it doesn't accept STN either).

![Screenshot_20220118-130527](https://user-images.githubusercontent.com/8660811/149973828-3d3bf6b9-72c7-4b8c-8ee1-d0c1bd9543b1.png)


Thanks.

**Comment by rtek1000** [comment, marked as answer]
https://github.com/orgs/stm32duino/discussions/1624#discussioncomment-1991496
